src/ruptures/detection/binseg.py

```diff
@@ -85,8 +84,6 @@ def _seg(self, n_bkps=None, pen=None, epsilon=None):
     def single_bkp(self, start, end):
         """Return the optimal breakpoint of [start:end] (if it exists)."""
         segment_cost = self.cost.error(start, end)
-        if np.isinf(segment_cost) and segment_cost < 0:  # if constant on segment
```

Comment:
I am not sure of the consequences of removing this. 

When using the `CostNormal` cost with the parameter `add_small_diag` set to `False`, with a constant signal, we still enter this condition (at least for relatively small signal sub-samples. For longer signal (>200 samples on my machine), covariance matrix is not 0.0 and we do not have segment_cost to `-np.inf`).

So this introduces a behaviour change in the case of constant signal with no added bias when the Normal cost returns `-np.inf` (which for some reasons is not always the case). 

In the end, it might be an ill case for which there is no good answer so we might as well remove this code section. I just thought it is worth mentioning it :)

Reply:
partially answered in https://github.com/deepcharles/ruptures/pull/198#discussion_r709163208